port/unix/omrsysinfo.c

```diff
@@ -7706,11 +7706,39 @@ omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallb
 alloc_failed:
 	callback_result = (uintptr_t)(intptr_t)OMRPORT_ERROR_SYSINFO_MEMORY_ALLOC_FAILED;
 	goto done;
-#elif defined(LINUX) /* defined(AIXPPC) */
-	uintptr_t callback_result = 0;
-	uintptr_t buffer_size = 4096;
+#elif defined(J9ZOS390) || defined(LINUX) /* defined(AIXPPC) */
+	uintptr_t callbackResult = 0;
+	uintptr_t bufferSize = 4096;
 	char *command = NULL;
-	DIR *dir = opendir("/proc");
+	DIR *dir = NULL;
+#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
+	uintptr_t outputBufferSize = bufferSize * 2;
+	char *utf8Command = (char *)portLibrary->mem_allocate_memory(
+			portLibrary,
+			outputBufferSize,
+			OMR_GET_CALLSITE(),
+			OMRMEM_CATEGORY_PORT_LIBRARY);
+	if (NULL == utf8Command) {
+		return OMRPORT_ERROR_SYSINFO_MEMORY_ALLOC_FAILED;
+	}
+
+#if defined(J9ZOS390)
```

Comment:
Lines 7725, 7727, 7729 and 7731 can safely be deleted; the condition is implied by line 7714.